src/timestamp.rs

```diff
@@ -0,0 +1,74 @@
+// LNP/BP Core Library implementing LNPBP specifications & standards
+// Written in 2020 by
+//     Dr. Maxim Orlovsky <orlovsky@pandoracore.com>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the MIT License
+// along with this software.
+// If not, see <https://opensource.org/licenses/MIT>.
+use chrono::{Utc, Local, DateTime, TimeZone};
+use std::ops::{Add, Sub};
+
+#[derive(Clone, Debug, PartialEq, Hash, Eq, PartialOrd, Ord)]
+pub struct Timestamp(pub u32);
+
+trait ToTimestamp {
+    fn to_timestamp(&self) -> Timestamp;
+}
+
+impl Timestamp {
+    pub fn from_seconds(seconds: u32) -> Self {
+        Timestamp(seconds)
+    }
+}
+
+impl Default for Timestamp {
+    fn default() -> Self {
+        Timestamp(Utc::now().timestamp() as u32)
+    }
+}
+
+impl Timestamp {
+    pub fn to_utc_datetime(&self) -> DateTime<Utc> {
+        Utc.timestamp(self.0 as i64, 0)
+    }
+
+    pub fn to_local_datetime(&self) -> DateTime<Local> {
+        Local.timestamp(self.0 as i64, 0)
+    }
+}
+
+
+impl ToTimestamp for DateTime<Utc> {
+    pub fn to_timestamp(&self) -> Timestamp {
+        Timestamp(self.timestamp() as u32)
+    }
+}
+
+
+impl ToTimestamp for DateTime<Local> {
+    pub fn to_timestamp(&self) -> Timestamp {
+        Timestamp(self.timestamp() as u32)
+    }
+}
+
+/// Add seconds to timestamp
+impl Add for Timestamp {
```

Comment:
These are correct, but default math traits panics on overflow. One of the reasons we have this newtype is the fact that we'd like to avoid that. Can you pls instead do custom functions after standard library (`checked_add`, `wrapping_add`, `overflowing_add`, `saturating_add` - and the same for sub). You can see how they work in https://doc.rust-lang.org/std/primitive.u32.html#method.checked_add

Reply:
Actually I think we may even use https://github.com/dpc/overflow-proof

Reply:
> 

What do you think would be the best way of proceed? I am thinking on this
```rust
  fn add(self, rhs: Timestamp) -> Timestamp {
    match self.0.checked_add(rhs.0) {
      Some(v) => Timestamp(v),
      None => Timestamp(0),
    }
  }
```